src/utils.ts

```diff
@@ -46,3 +48,92 @@ export function setContext(kubeconfig: string): string {
    kc.setCurrentContext(context)
    return kc.exportConfig()
 }
+
+/**
+ * Sets the context by pulling kubeconfig via az command
+ * @param admin Boolean for determining whether or not user is admin
+ * @param kubeconfigPath Path to place kubeconfig
+ * @returns Promise for the resulting exitCode number from running az command
+ */
+export async function azSetContext(
+   admin: boolean,
+   kubeconfigPath: string
+): Promise<number> {
+   const AZ_TOOL_NAME = 'az'
+   const AZ_USER_AGENT_ENV = 'AZURE_HTTP_USER_AGENT'
+   const AZ_USER_AGENT_ENV_PS = 'AZUREPS_HOST_ENVIRONMENT'
+   const originalAzUserAgent = process.env[AZ_USER_AGENT_ENV] || ''
+   const originalAzUserAgentPs = process.env[AZ_USER_AGENT_ENV_PS] || ''
+
+   try {
+      // set az user agent
+      core.exportVariable(AZ_USER_AGENT_ENV, getUserAgent(originalAzUserAgent))
+      core.exportVariable(
+         AZ_USER_AGENT_ENV_PS,
+         getUserAgent(originalAzUserAgentPs)
+      )
+
+      // get inputs
+      const resourceGroupName: string = core.getInput('resource-group', {
+         required: true
+      })
+      const clusterName: string = core.getInput('cluster-name', {
+         required: true
+      })
+      const subscription: string = core.getInput('subscription') || ''
+      const cmd = [
+         'aks',
+         'get-credentials',
+         '--resource-group',
+         resourceGroupName,
+         '--name',
+         clusterName,
+         '-f',
+         kubeconfigPath
+      ]
+
+      // check az tools
+      const azPath = await io.which(AZ_TOOL_NAME, false)
+      if (!azPath)
+         throw Error(
+            'Az cli tools not installed. You must install them before running this action with the aks-set-context flag'
+         )
+
+      if (subscription) cmd.push('--subscription', subscription)
+      if (admin) cmd.push('--admin')
+
+      const exitCode = await exec.exec(AZ_TOOL_NAME, cmd)
+
+      return exitCode
+   } catch (e) {
+      throw e
+   } finally {
+      core.exportVariable(AZ_USER_AGENT_ENV, originalAzUserAgent)
+      core.exportVariable(AZ_USER_AGENT_ENV_PS, originalAzUserAgentPs)
+   }
+}
+
+/**
+ * Uses kubelogin to convert kubeconfig to exec credential plugin format
+ * @param exitCode ExitCode from az command execution to obtain kubeconfig
+ */
+export async function kubeLogin(exitCode: number): Promise<void> {
+   const KUBELOGIN_TOOL_NAME = 'kubelogin'
+   const kubeloginCmd = ['convert-kubeconfig', '-l', 'azurecli']
+
+   const kubeloginExitCode = await exec.exec(KUBELOGIN_TOOL_NAME, kubeloginCmd)
+   if (kubeloginExitCode !== 0)
```

Comment:
I might be missing something but is the exitCode parameter necessary? If it's non-zero, doesn't the statement at line 32 throw and set the action to fail? Moreover, wouldn't you want to return/keep track of kubeloginExitCode since that's the exit status for the kubelogin command?